### PR TITLE
Add global and init section and document behaviour in a new test

### PR DESCRIPTION
**Pull request by daniel-grumberg**

### CODE REVIEW

File: src/Syntax.hs [hunk 1]
```diff
@@ -38,11 +40,15 @@ data Modifier = Star deriving (Show, Eq, Ord)
...
+type Global = Maybe Code
+
+type Init = Maybe Code
+
 type Mapping = (Syscall, Rhs)
 
 type Begin = [Syscall]
 
-type IR = ([Include], (Begin, EndHook, Skips), [Mapping] , [Rhs])
+type IR = (([Include], Global, Init), (Begin, EndHook, Skips), [Mapping], [Rhs])
```
Comment (luisggpina) [inline src/Syntax.hs]: For legibility, perhaps we could give `([Include], Global, Init)` a name.  Like: `type Preamble = ([Include], Global, Init)`

File: test/FrontEnd.hs [hunk 2]
```diff
@@ -44,9 +44,18 @@ arguments = parens . (commaList arg)
   where arg Any     = text "_"
         arg (Arg (Name s)) = text s
 
+rendCode :: Code -> Doc
+rendCode (Code s) = doubleBraces $ text s
+  where doubleBraces p = text"{{" <> p <> text "}}"
```
Comment (luisggpina) [inline test/FrontEnd.hs]: Missing space after `text"{{"`, should be `text "{{"`

Comment (luisggpina) [inline test/FrontEnd.hs]: Why name `rendCode`?  `parseCode` sounds more natural to me

File: src/CodeGen.hs [hunk 3]
```diff
@@ -182,21 +184,23 @@ declareFps
...
-initFunc mappings [] skips lhsNothing
+initFunc :: [Mapping] -> [Syscall] -> Skips -> [Rhs] -> Init -> Doc
+initFunc mappings [] skips lhsNothing ini
   = initFps mappings <$$>
     text "void %init_div%() {" <$$>
     indent 2 (text "init_fps();") <$$>
     initSkips skips <>
-    initLHSNothing lhsNothing <>
+    initLHSNothing lhsNothing <$$>
+    maybe empty (\(Code s) -> text s) ini<>
```
Comment (luisggpina) [inline src/CodeGen.hs]: Missing space after `ini<>`, should be `ini <>`

File: src/Syntax.hs (hunk 1 above)
Comment (daniel-grumberg) [inline src/Syntax.hs]: Agreed, the `IR` and `Syscall` types are getting too large I will create a further PR down the line to use record syntax for them to simplify things a bit.

File: test/FrontEnd.hs (hunk 2 above)
Comment (daniel-grumberg) [inline test/FrontEnd.hs]: This is the pretty printer needed for the fuzz testing for a IR Code block, so rend (for render) makes more sense.

**Comment by daniel-grumberg** [comment]
Fixed the whitespace problems

**Comment by luisggpina** [comment]
Fixes #17

**Pull request merged**